### PR TITLE
fix golint errors for `pkg/kubelet/dockershim`

### PR DESCRIPTION
**Pull request by zvier**
Signed-off-by: zvier <zvier20@gmail.com>

**What type of PR is this?**
 /kind cleanup

**What this PR does / why we need it**:
Fix golint issues for the package of `pkg/kubelet/dockershim`

**Which issue(s) this PR fixes**:

Ref #68026

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:
NONE

### CODE REVIEW

**Comment by Joseph-Irving** [comment]
Hi @zvier you must have signed the CLA to contribute to Kubernetes, see this comment https://github.com/kubernetes/kubernetes/pull/91181#issuecomment-629821535, if you're having difficulties signing, you can try the helpdesk

**Comment by zvier** [comment]
cla ok

**Comment by zvier** [comment]
> Hi @zvier you must have signed the CLA to contribute to Kubernetes, see this comment [#91181 (comment)](https://github.com/kubernetes/kubernetes/pull/91181#issuecomment-629821535), if you're having difficulties signing, you can try the helpdesk

hello，I'm signed the CLA follow the above steps, and checked my github account email infomation is consistent with my CNCF account infomation. But it seems not work, is there any problem?

**Comment by Joseph-Irving** [comment]
if you've followed all the step and it's still not working I'd recommend trying the helpdesk at https://support.linuxfoundation.org/

**Comment by zvier** [comment]
I signed it

**Comment by zvier** [comment]
I signed it

**Comment by zvier** [comment]
I signed it

**Comment by zvier** [comment]
> if you've followed all the step and it's still not working I'd recommend trying the helpdesk at https://support.linuxfoundation.org/

Thank you, I have succeeded～

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/docker_checkpoint.go:37]: Can you share a little more about the name `ContainerCheckpoint` and how its the most appropriate replacement for `DockershimCheckpoint`?

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/docker_checkpoint.go:78]: Nit: does `int` mean `initializes` here? I think using the full word will help make this more readable.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/docker_checkpoint.go:88]: Small nit: For `MarshalCheckpoint`, `UnmarshalCheckpoint`, and `GetData`, I think we want `MarshalCheckpoint encodes`, `UnmarshalCheckpoint decodes`, and `GetData gets`.

File: pkg/kubelet/dockershim/exec.go
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/exec.go:63]: Nit: `s/execute/executes`

File: pkg/kubelet/dockershim/helpers.go
Comment (mattjmcnaughton) [inline pkg/kubelet/dockershim/helpers.go:292]: Nit: I think that the following might be more readable...

```
rmErr := client.RemoveContainer...
if rmEr == nil {
}
...
```

**Review by mattjmcnaughton — COMMENTED** [review]
Thanks for addressing these errors. Just a couple of small comments from me, but overall, its on a great path.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (zvier) [reply pkg/kubelet/dockershim/docker_checkpoint.go:37]: The golint error here is:
```
docker_checkpoint.go:35:6: type name will be used as dockershim.DockershimCheckpoint by other packages, and that stutters; consider calling this Checkpoint
```

Because there is already a PodSandboxCheckpoint implement this interface. I think ContainerCheckpoint would be more general. May be a DockerCheckpoint, KataCheckpoint or any other type container's Checkpoint will implement this interface in the future.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (zvier) [reply pkg/kubelet/dockershim/docker_checkpoint.go:78]: Sorry, it's a typo, I'll fix it

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (MHBauer) [inline pkg/kubelet/dockershim/docker_checkpoint.go:99]: wheather -> whether

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (MHBauer) [inline pkg/kubelet/dockershim/docker_checkpoint.go:78]: init -> inits

File: pkg/kubelet/dockershim/doc.go
Comment (MHBauer) [inline pkg/kubelet/dockershim/doc.go:19]: I would drop "does all of the work to implement" and replace with "package dockershim implements a container runtime interface"

contaienr -> container

File: pkg/kubelet/dockershim/helpers_linux.go
Comment (MHBauer) [inline pkg/kubelet/dockershim/helpers_linux.go:37]: return -> returns

**Review by MHBauer — COMMENTED** [review]
nits but looking good.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (mattjmcnaughton) [reply pkg/kubelet/dockershim/docker_checkpoint.go:37]: Could `ProcessContainerCheckpoint` be even more specific? Or does that just increase confusion?

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (zvier) [reply pkg/kubelet/dockershim/docker_checkpoint.go]: Thank you for you comment， I will be careful for my spelling.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (zvier) [reply pkg/kubelet/dockershim/docker_checkpoint.go:37]: For this interface name, I think ContainerCheckpoint may be more suitable. There exist an implement of PodSandboxCheckpoint, and the podsandbox is just a special container. But say it's a process container sounds not good ?

**Comment by zvier** [comment]
Update my emal address

**Review by MHBauer — APPROVED** [review]
/ok-to-test
/lgtm

definitely need to pick a different approver.

File: pkg/kubelet/dockershim/docker_checkpoint.go
Comment (Joseph-Irving) [inline pkg/kubelet/dockershim/docker_checkpoint.go:105]: `infomation` should be `information`

**Comment by zvier** [comment]
> New changes are detected. LGTM label has been removed.

Update for fix three Errors of golint testcase report :
1.pkg/kubelet/dockershim/docker_stats_windows.go:36:2: don't use underscores in Go names; var hcsshim_container should be hcsshimContainer
2.pkg/kubelet/dockershim/helpers_unsupported.go:30:1: exported function DefaultMemorySwap should have comment or be unexported
3.pkg/kubelet/dockershim/helpers_windows.go:35:1: exported function DefaultMemorySwap should have comment or be unexported

**Comment by zvier** [comment]
@yifan-gu @ixdy PTAL

**Review by SergeyKanzhelev — APPROVED** [review]

**Comment by zvier** [comment]
Hi @dims , could you approve the PR if it looks good to you?

**Comment by dims** [comment]
/assign @dashpole @sjenning 

i don't have `/approve` rights :)

**Comment by SergeyKanzhelev** [comment]
@dims your approval is needed for hack/OWNERS.

**Comment by SergeyKanzhelev** [comment]
@dims this PR intended to eliminate this need: https://github.com/kubernetes/kubernetes/pull/92617 please take a look

**Comment by dims** [comment]
/approve

whoops!

**Pull request merged**
